### PR TITLE
Move remaining CUB policy hubs to tuning headers

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Fixes: #3097

### CODE REVIEW

File: cub/cub/device/dispatch/dispatch_merge_sort.cuh [hunk 1]
```diff
@@ -327,61 +328,13 @@ __launch_bounds__(
...
-};
-
 template <typename KeyInputIteratorT,
           typename ValueInputIteratorT,
           typename KeyIteratorT,
           typename ValueIteratorT,
           typename OffsetT,
           typename CompareOpT,
-          typename SelectedPolicy = DeviceMergeSortPolicy<KeyIteratorT>>
+          typename SelectedPolicy = detail::merge_sort::policy_hub<KeyIteratorT>>
```
Comment (miscco) [inline cub/cub/device/dispatch/dispatch_merge_sort.cuh:337]: Should we rename this to PolicyHub

**Review by miscco — COMMENTED** [review]
Minor question about SelectedPolicy vs PolicyHub

**Review by miscco — APPROVED** [review]

File: cub/cub/device/dispatch/dispatch_merge_sort.cuh (hunk 1 above)
Comment (bernhardmgruber) [reply cub/cub/device/dispatch/dispatch_merge_sort.cuh:337]: Yes, it's always PolicyHub. Will rename in a separate PR.

Comment (bernhardmgruber) [reply cub/cub/device/dispatch/dispatch_merge_sort.cuh:337]: Addressed in #3142

**Pull request merged**